Plutarch/Prelude.hs

```diff
@@ -13,17 +14,143 @@ module Plutarch.Prelude (
   (#),
   plet,
   pinl,
-  pcon,
-  pmatch,
   pto,
   pfix,
   Type,
   S,
   PType,
+  PlutusType,
+  PCon (pcon),
+  PMatch (pmatch),
+
+  -- * Integers and integer utilities
+  PInteger,
+  PIntegral (pdiv, pmod, pquot, prem),
+
+  -- * Rational numbers and utilities
+  PRational,
+  pnumerator,
+  pdenominator,
+  pfromInteger,
+  pround,
+
+  -- * Booleans and boolean functions
+  PBool (..),
+  PEq ((#==)),
+  POrd ((#<=), (#<)),
+  pif,
+  pnot,
+  (#&&),
+  (#||),
+
+  -- * Bytestrings and bytestring utilities
+  PByteString,
+  phexByteStr,
+  pconsBS,
+  psliceBS,
+  plengthBS,
+  pindexBS,
+
+  -- * String and string utilities
+  PString,
+  pencodeUtf8,
+  pdecodeUtf8,
+
+  -- * Unit type and utilities
+  PUnit (..),
+
+  -- * Common list typeclass and utilities
+  PListLike (PElemConstraint, pelimList, pcons, pnil, phead, ptail, pnull),
+  PIsListLike,
+  plistEquals,
+  pelem,
+  plength,
+  ptryIndex,
+  pdrop,
+  psingleton,
+  pconcat,
+  pzipWith,
+  pzipWith',
+  pzip,
+  pmap,
+  pfilter,
+  precList,
+  pfoldr,
+  pfoldrLazy,
+  pfoldl,
+  pall,
+  pany,
+
+  -- * Scott encoded list type
+  PList (..),
+
+  -- * Scott encoded maybe type and utilities
+  PMaybe (..),
+
+  -- * Scott encoded either type and utilities
+  PEither (..),
+
+  -- * Scott encoded pair type and utilities
+  PPair (..),
+
+  -- * Builtin types and utilities
+  PData (..),
+  pfstBuiltin,
+  psndBuiltin,
+  PBuiltinPair,
+  PBuiltinList (..),
+  PIsData (pfromData, pdata),
+  PAsData,
+  ppairDataBuiltin,
+  PBuiltinMap,
+
+  -- * DataRepr and related functions
+  PDataFields,
+  PDataRecord,
+  PDataSum,
+  PIsDataRepr,
+  pfield,
+  pletFields,
+
+  -- * Tracing
+  ptrace,
+  ptraceIfFalse,
+  ptraceIfTrue,
+  ptraceError,
+
+  -- * Cryptographic hashes and signatures
+  psha2_256,
+  psha3_256,
+  pverifySignature,
+
+  -- * Converstion between Plutarch terms and Haskell types
+  pconstant,
+  plift,
+  plift',
+  LiftError,
```

Comment:
We don't need this.

Reply:
I assume the `LiftError` export should be removed *alongside* `plift'` then.

Reply:
Yeah.

Reply:
Removed.